src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -273,9 +259,13 @@ public override int Read(byte[] array, int offset, int count)
                 Debug.Assert(_inflater.NeedsInput(), "We can only run into this case if we are short of input");
 
                 int bytes = _stream.Read(_buffer, 0, _buffer.Length);
-                if (bytes == 0)
+                if (bytes <= 0)
                 {
-                    break;      //Do we want to throw an exception here?
+                    break;
+                }
+                else if (bytes > _buffer.Length)
+                {
+                    throw new InvalidDataException(SR.GenericInvalidData);
```

Comment:
same below at 381.  What exception were we throwing later on?  This could be a compat concern, though unlikely.